### PR TITLE
Remove wait for ajax calls

### PR DESCRIPTION
**Pull request by skshetry**
## Description

We were getting CI failures during resetting/setting the password regarding the `jQuery` not getting found when we evaluate script, and only on Firefox.
```
Firefox WebDriver\Exception\JavaScriptError: ReferenceError: jQuery is not defined
Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T19:05:32.194Z'
      #1 /drone/src/vendor-bin/behat/vendor/instaclick/php-webdriver/lib/WebDriver/AbstractWebDriver.php(218): WebDriver\AbstractWebDriver->curl('POST', '/execute', Array)
      #2 /drone/src/vendor-bin/behat/vendor/instaclick/php-webdriver/lib/WebDriver/Container.php(224): WebDriver\AbstractWebDriver->__call('execute', Array)
      #3 /drone/src/vendor-bin/behat/vendor/behat/mink-selenium2-driver/src/Selenium2Driver.php(949): WebDriver\Container->__call('execute', Array)
      #4 /drone/src/vendor-bin/behat/vendor/behat/mink/src/Session.php(336): Behat\Mink\Driver\Selenium2Driver->evaluateScript('return (\n\t\t\t\tfu...')
      #5 /drone/src/tests/acceptance/features/lib/OwncloudPage.php(555): Behat\Mink\Session->evaluateScript('(\n\t\t\t\tfunction ...')
      #6 /drone/src/tests/acceptance/features/lib/OwncloudPage.php(595): Page\OwncloudPage->waitForAjaxCallsToStart(Object(Behat\Mink\Session))
      #7 /drone/src/tests/acceptance/features/lib/LoginPage.php(219): Page\OwncloudPage->waitForAjaxCallsToStartAndFinish(Object(Behat\Mink\Session))
      #8 /tmp/ProxyManagerGeneratedProxy__PM__PageLoginPageGenerated6312256563dff9947346b9792a817253.php(94): Page\LoginPage->resetThePassword('aVeryLongPasswo...', 'aVeryLongPasswo...', Object(Behat\Mink\Session))
      #9 /drone/src/tests/acceptance/features/bootstrap/WebUILoginContext.php(557): ProxyManagerGeneratedProxy\__PM__\Page\LoginPage\Generated6312256563dff9947346b9792a817253->resetThePassword('aVeryLongPasswo...', 'aVeryLongPasswo...', Object(Behat\Mink\Session))
      #10 [internal function]: WebUILoginContext->theUserResetsThePasswordWithSameConfirmationToUsingTheWebui('aVeryLongPasswo...')
```

Looking into it, there were no waits of any sorts, so, I added wait for the reset/set password page to load via #35115. But, this did not solve any issues. So, next thing that came to mind, was that the jQuery might not have been available at the time we evaluate the script (added via #35167) as the page must still be loading. But, this was only partially true. 

The reset/set password form was submitted, and then, the check for `ajax` calls were done. When we submit the form, and the client-side validation checks out, the user is redirected to the login page. So, when that `ajax` call ran, the form was submitted and the login page was getting loaded making the `jQuery` not available. This did escaped the `waitTillJqueryIsLoaded` as it was still on the reset/set password page when that function ran. The `waitForAjaxCallsToStartAndFinish` is not useful here anyway (as page reload happens to different page, so, before the timeout is reached, another page might already be getting loaded).

The ajax calls do happen if `user_management` and `password_policy` app is installed. But, later page reloads, due to which it won't be effective/useful to check for `ajax` calls, and the responsibility should be shifted to later steps to handle. I'll make a test PR for `user_management` and `password_policy`. Let, this PR stay here for few days (and, run multiple times) to ensure this does not happen again.

The PR #35115 and #35167, even though did not solve the issue, are, I believe necessary to reduce the flakiness.

## Proposition
We know, `waitForAjaxCallsToStartAndFinish` cannot be trusted (as there could be `XHR` calls due to other reasons, for example, pulling new notifications) and results in flaky tests. So, I'd propose we discourage the use of it, and encourage tests of other reactions/interactions in the page or, even api calls to reduce this. We could  start by echoing a message regarding this, and to suggest the use of the function iff there is no other way.

## Related Issue

- Trying to fix CI issue in Firefox.

## Motivation and Context

[Acceptance tests failure](https://drone.owncloud.com/owncloud/core/17168/903).

## How Has This Been Tested?

Needs to be tested thoroughly in CI.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by skshetry** [comment]
Backport for `stable10` on #35172

**Comment by skshetry** [comment]
See PR for `password_policy`: owncloud/password_policy#221

**Comment by skshetry** [comment]
The `user_management` app should be tested on `stable10` by #35172

File: tests/acceptance/features/lib/OwncloudPage.php [hunk 1]
```diff
@@ -60,22 +60,20 @@ class OwncloudPage extends Page {
 	private function waitTillJqueryIsLoaded(
 		Session $session,
 		int $timeout_msec = STANDARD_UI_WAIT_TIMEOUT_MILLISEC
-	): void {
```
Comment (skshetry) [inline tests/acceptance/features/lib/OwncloudPage.php]: I was wrong. The return type declarations was introduced in phpv7.0, but `void` return type was only introduced in [phpv7.1](https://www.php.net/manual/en/migration71.new-features.php#migration71.new-features.void-functions)

**Review by individual-it — APPROVED** [review]

**Review by individual-it — APPROVED** [review]

**Review by individual-it — DISMISSED** [review]
waiting till password policy is passing https://github.com/owncloud/password_policy/pull/221

**Review by phil-davis — APPROVED** [review]

**Pull request merged**